lib/sedna/core/lifelong_learning/lifelong_learning.py

```diff
@@ -165,95 +224,113 @@ def train(self, train_data,
+        task_index = self.cloud_knowledge_management.update_kb(
+            task_index_url, self.kb_server)
+        res.update(unseen_res)
 
         task_info_res = self.estimator.model_info(
-            self.config.task_index,
+            task_index,
             relpath=self.config.data_path_prefix)
         self.report_task_info(
             None, K8sResourceKindStatus.COMPLETED.value, task_info_res)
         self.log.info(f"Lifelong learning Train task Finished, "
-                      f"KB idnex save in {self.config.task_index}")
+                      f"KB index save in {task_index}")
         return callback_func(self.estimator, res) if callback_func else res
 
     def update(self, train_data, valid_data=None, post_process=None, **kwargs):
-        return self.train(
-            train_data=train_data,
-            valid_data=valid_data,
-            post_process=post_process,
-            action="update",
-            **kwargs
-        )
+        """
+        fit for update the knowledge based on incremental data.
+
+        Parameters
+        ----------
+        train_data : BaseDataSource
+            Train data, see `sedna.datasources.BaseDataSource` for more detail.
+        valid_data : BaseDataSource
+            Valid data, BaseDataSource or None.
+        post_process : function
+            function or a registered method, callback after `estimator` train.
+        kwargs : Dict
+            parameters for `estimator` training, Like:
+            `early_stopping_rounds` in Xgboost.XGBClassifier
+
+        Returns
+        -------
+        train_history : object
+        """
+        callback_func = None
+        if post_process is not None:
+            callback_func = ClassFactory.get_cls(
+                ClassType.CALLBACK, post_process)
+
+        task_index_url = self.get_parameters(
+            "CLOUD_KB_INDEX", self.cloud_knowledge_management.task_index)
+        index_url = self.cloud_knowledge_management.local_task_index_url
+        FileOps.download(task_index_url, index_url)
+
+        unseen_sample_re_recognition = ClassFactory.get_cls(
+            ClassType.UTD, self.unseen_sample_re_recognition["method"])(
+            index_url, **self.unseen_sample_re_recognition_param)
+
+        seen_samples, unseen_samples = unseen_sample_re_recognition(train_data)
+
+        # TODO: retrain temporarily
+        # historical_data = self._fetch_historical_data(index_url)
+        # seen_samples.x = np.concatenate(
+        #     (historical_data.x, seen_samples.x, unseen_samples.x), axis=0)
+        # seen_samples.y = np.concatenate(
+        #     (historical_data.y, seen_samples.y, unseen_samples.y), axis=0)
+
+        seen_samples.x = np.concatenate(
+            (seen_samples.x, unseen_samples.x), axis=0)
+        seen_samples.y = np.concatenate(
+            (seen_samples.y, unseen_samples.y), axis=0)
+
+        task_update_decision = ClassFactory.get_cls(
```

Comment:
if `task_update_decision` is callable, then skip `ClassFactory.get_cls` method and set task index instead.